### ISSUE TITLE
'str' object has no attribute 'append'

### ISSUE THREAD

**Issue by mDafox** [Rudy-Tmc/backupAndRestoreInsightCloud#2]
We're trying to use your scripts to backup an Object scheme to import in our Sandbox environment.
After some succesful tests with test schemes we now want to do the same with our main scheme but stumble on the following error

```
2022-07-06 16:15:08,430 root         INFO     [28] -----------Start of Run-----------
2022-07-06 16:15:09,312 root         INFO     [74] Start backup of:
2022-07-06 16:15:09,312 root         INFO     [76]    IAM [AOS]
2022-07-06 16:15:09,313 root         INFO     [82]    - objectschema
2022-07-06 16:15:09,662 root         INFO     [87]    - object schema properties
2022-07-06 16:15:10,117 root         INFO     [92]    - global referencetypes
2022-07-06 16:15:10,469 root         INFO     [97]    - global statustypes
2022-07-06 16:15:10,857 root         INFO     [103]    - schema referencetypes
2022-07-06 16:15:11,308 root         INFO     [108]    - schema statustypes
2022-07-06 16:15:11,676 root         INFO     [114]    - objecttypes [10]
2022-07-06 16:15:11,677 root         INFO     [120] - 'Team' [1/10]:
2022-07-06 16:15:12,145 root         INFO     [123]      - attributes [11]
2022-07-06 16:15:12,856 root         INFO     [128]      - object types [19]
2022-07-06 16:15:13,775 root         INFO     [142]         - data
2022-07-06 16:15:14,976 root         INFO     [154]         - history
2022-07-06 16:15:15,774 root         INFO     [166]         - comments
2022-07-06 16:15:15,774 root         INFO     [120] - 'App Journeys' [2/10]:
2022-07-06 16:15:16,105 root         INFO     [123]      - attributes [10]
2022-07-06 16:15:18,275 root         INFO     [128]      - object types [60]
2022-07-06 16:15:20,055 root         INFO     [142]         - data
2022-07-06 16:15:22,526 root         INFO     [154]         - history
2022-07-06 16:15:24,166 root         INFO     [166]         - comments
2022-07-06 16:15:24,167 root         INFO     [120] - 'Application' [3/10]:
2022-07-06 16:15:24,524 root         INFO     [123]      - attributes [18]
2022-07-06 16:15:26,971 root         INFO     [128]      - object types [51]
2022-07-06 16:15:28,662 root         INFO     [142]         - data
2022-07-06 16:15:31,172 root         INFO     [154]         - history
2022-07-06 16:15:32,747 root         INFO     [166]         - comments
2022-07-06 16:15:32,748 root         INFO     [120] - 'Web Journeys' [4/10]:
2022-07-06 16:15:33,103 root         INFO     [123]      - attributes [10]
2022-07-06 16:15:33,599 root         INFO     [128]      - object types [9]
2022-07-06 16:15:34,076 root         INFO     [142]         - data
2022-07-06 16:15:34,741 root         INFO     [154]         - history
2022-07-06 16:15:35,218 root         INFO     [166]         - comments
2022-07-06 16:15:35,218 root         INFO     [120] - 'Person' [5/10]:
2022-07-06 16:15:35,655 root         INFO     [123]      - attributes [7]
2022-07-06 16:15:43,157 root         INFO     [128]      - object types [276]
2022-07-06 16:15:52,153 root         ERROR    [175] Unhandled error
'str' object has no attribute 'append'
Traceback (most recent call last):
  File "C:\temp\backupAndRestoreInsightCloud-main\backupInsight.py", line 139, in <module>
    objectId, objectData = future.result()
  File "C:\Users\manuv\AppData\Local\Programs\Python\Python310\lib\concurrent\futures\_base.py", line 439, in result
    return self.__get_result()
  File "C:\Users\manuv\AppData\Local\Programs\Python\Python310\lib\concurrent\futures\_base.py", line 391, in __get_result
    raise self._exception
  File "C:\Users\manuv\AppData\Local\Programs\Python\Python310\lib\concurrent\futures\thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\temp\backupAndRestoreInsightCloud-main\backupInsight.py", line 31, in getObjectData
    objectData = myInsight.getObjectData(object)
  File "C:\temp\backupAndRestoreInsightCloud-main\insight.py", line 857, in getObjectData
    attributeValue.append(value['displayValue'])
AttributeError: 'str' object has no attribute 'append'
2022-07-06 16:15:52,156 root         INFO     [178] ------------End of Run------------

```

We looked at what could be different between the object types that did succeed (such as Team) but so far have not found differences that seem to indicate the cause of the issue.
Also tried with different api keys so far

**Comment by Rudy-Tmc** [comment]
Hi Manu,
Two questions, did you use the latest version (2.0) and could you run the script in DEBUG mode (change line 18 from INFO to DEBUG). This will generate an backup.log which contains more info, could you send that to me?

Note that API keys are not the problem.

**Comment by Rudy-Tmc** [comment]
If you run it in DEBUG mode you can also see which Person object fails.

**Comment by mDafox** [comment]
> Hi Manu, Two questions, did you use the latest version (2.0) and could you run the script in DEBUG mode (change line 18 from INFO to DEBUG). This will generate an backup.log which contains more info, could you send that to me?
> 
> Note that API keys are not the problem.

Could I send the logfile to you in private somehow? This Jira instance is for one of our customers and I don't want to risk posting anything public about it

**Comment by mDafox** [comment]
> I understand, I will treat it very carefully and deleted it when this issue is solved/clear.
> 
> You can use this obfuscated address: [ivnw9e54@duck.com](mailto:ivnw9e54@duck.com) to send it to me

Sent the log file!

**Comment by Rudy-Tmc** [comment]
Got him

**Comment by Rudy-Tmc** [comment]
Fixed in release 2.1
